Tests/LibC/TestFenv.cpp

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2023, kleines Filmröllchen <filmroellchen@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibTest/TestCase.h>
+#include <fenv.h>
+
+// TODO: Add tests for floating-point exception management.
+
+// This error happens even with float literals; e.g. -1.099999f becomes -1.099998f.
+// However, it is enough to distinguish the different rounding errors we test for.
+static constexpr double acceptable_float_error = 0.000001;
+
+TEST_CASE(float_round_up)
+{
+    fesetround(FE_UPWARD);
+    EXPECT_APPROXIMATE_WITH_ERROR(0.1f + 0.2f, 0.3f, acceptable_float_error);
```

Comment:
Can we use precise comparison instead? Since our formatter for float and double is roundtrip safe, it shouldn't be hard to obtain correct values to compare to.

Reply:
Not sure how that holds up across architectures, but I don't think this is a problem.

Reply:
Huh? Floating point rounding behavior is specified in IEEE 754 just like everything else regarding them.